airflow/models/__init__.py

```diff
@@ -2452,9 +2454,35 @@ def render_template_from_field(self, attr, content, context, jinja_env):
                 k: rt("{}[{}]".format(attr, k), v, context)
                 for k, v in list(content.items())}
         else:
-            result = content
+            result = self._render_template_object_recursively(content, context)
         return result
 
+    def _render_template_object_recursively(self, content, context):
+        if id(content) not in self._rendered_template_object_ids:
+            self._rendered_template_object_ids.add(id(content))
+            try:
+                content_attributes = vars(content)
+            except TypeError:
+                # content has no __dict__ (e.g.: could be an int)
+                return content
+
+            for attribute, value in content_attributes.items():
+                try:
+                    # Best effort approach here:
+                    # try to render attribute value as a template,
+                    # and try to set this attribute with the rendering result
+                    # It may fail sometimes
+                    # (e.g.: UUID has an int attribute but is immutable; see UUID.__settattr__ )
+                    # In case of failure, just let this attribute unchanged.
+                    setattr(content,
+                            attribute,
+                            self.render_template(attribute, value, context))
+                except Exception:
```

Comment:
I would strongly prefer more granular error handling here. Specifically, any exception raised from `render_template` should be propagated out.

Reply:
My intention was a best effort approach: try to render inner attributes as templates, or else keep attributes unchanged.

e.g: when traversing a `UUID` attribute, `render_template` succeeds, but `setattr` raises an exception (`UUID` is immutable)

But you have a good point: if my inner template expression is wrong, I would like the exception to be be propagated out, so I can understand the issue.
I could just put the `render_template` call outside try statement?

I will add some test cases right away

Reply:
Yes, that would be acceptable. I'd also consider logging any exception at least as a warning.